### PR TITLE
[linting_tool] Implement exporting profiles

### PR DESCRIPTION
**Pull request by abd99**
/cc @domesticmouse

### CODE REVIEW

File: experimental/linting_tool/lib/pages/saved_lints_page.dart [hunk 1]
```diff
@@ -45,7 +45,7 @@ class SavedLintsPage extends StatelessWidget {
               ),
               itemCount: profilesStore.savedProfiles.length,
               cacheExtent: 5,
-              itemBuilder: (context, index) {
+              itemBuilder: (itemBuilderContext, index) {
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/pages/saved_lints_page.dart:48]: I'm curious why the `context` rename here

File: experimental/linting_tool/lib/repository/api_provider.dart [hunk 2]
```diff
@@ -5,14 +5,15 @@
...
+import 'package:yaml/yaml.dart';
 
 class APIProvider {
   final _baseURL = 'https://dart-lang.github.io/linter';
   final http.Client httpClient;
   APIProvider(this.httpClient);
 
   Future<List<Rule>> getRulesList() async {
-    http.Response response =
+    var response =
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/repository/api_provider.dart:16]: `final`? =)

**Review by domesticmouse — APPROVED** [review]

File: experimental/linting_tool/lib/pages/saved_lints_page.dart (hunk 1 above)
Comment (abd99) [reply experimental/linting_tool/lib/pages/saved_lints_page.dart:48]: I'm passing the `context` to a `ScaffoldMessenger`(to show a SnackBar) later in the code. If it isn't renamed, this context gets passed which causes the following error:
```
Unhandled Exception: Looking up a deactivated widget's ancestor is unsafe.
At this point the state of the widget's element tree is no longer stable.
To safely refer to a widget's ancestor in its dispose() method, save a reference to the ancestor by calling dependOnInheritedWidgetOfExactType() in the widget's didChangeDependencies() method.
```

File: experimental/linting_tool/lib/repository/api_provider.dart (hunk 2 above)
Comment (abd99) [reply experimental/linting_tool/lib/repository/api_provider.dart:16]: Sure.

Comment (abd99) [reply experimental/linting_tool/lib/repository/api_provider.dart]: I wonder why we don't have lint rules enabled to pick this up? Should I create a new PR for this?

File: experimental/linting_tool/lib/pages/saved_lints_page.dart (hunk 1 above)
Comment (domesticmouse) [reply experimental/linting_tool/lib/pages/saved_lints_page.dart:48]: I'll look more closely at this after it is landed.

File: experimental/linting_tool/lib/repository/api_provider.dart (hunk 2 above)
Comment (domesticmouse) [reply experimental/linting_tool/lib/repository/api_provider.dart]: I suspect we are missing a lint rule. What lints are setup for this project?

Comment (abd99) [reply experimental/linting_tool/lib/repository/api_provider.dart]: I have used the default lints that are used in most of the projects in the repo.
[Example](https://github.com/flutter/samples/blob/master/veggieseasons/analysis_options.yaml)

**Pull request merged**